File: Leetcode_重刷紀錄/0023.Problems2352. Equal Row and Column Pairs/Equal Row and Column Pairs.java
//別人的程式碼
class Solution {
    public int equalPairs(int[][] grid) {
        Map<List<Integer> , Integer> data = new HashMap<>();
        int ans = 0;

        //storing the row elements
        for(int i=0;i<grid.length;i++){
            List<Integer> temp = new ArrayList<>();
            for(int j=0;j<grid.length;j++){
                temp.add(grid[i][j]);
            }
            data.put(temp, data.getOrDefault(temp, 0)+1);
        }

       
        //storing the col elements
        for(int j=0;j<grid.length;j++){
            List<Integer> col = new ArrayList<>();
            for(int i=0;i<grid.length;i++){
                col.add(grid[i][j]);
            }
          
            if(data.containsKey(col)){
                ans+=data.get(col);
            }
        }
        return ans;
    }
}

File: Leetcode_重刷紀錄/0023.Problems2352. Equal Row and Column Pairs/Equal Row and Column Pairs.py
class Solution:
    def equalPairs(self, grid: List[List[int]]) -> int:
       # 計算行的出現次數
        row_counts = {}
        for row in grid:
            row_tuple = tuple(row)  # 將每行轉換為元組以便哈希
            row_counts[row_tuple] = row_counts.get(row_tuple, 0) + 1
        
        count = 0
        n = len(grid)
        # 檢查每一列是否在行中出現
        for col in range(n):
            col_tuple = tuple(grid[row][col] for row in range(n))  # 建構列的元組
            count += row_counts.get(col_tuple, 0)  # 如果列在行中出現，則增加對應的次數
        
        return count